src/main/java/org/sunbird/hubservices/daoimpl/GraphDao.java

```diff
@@ -25,292 +14,279 @@
-
-				StringBuilder linkNthLevel = new StringBuilder();
-				for (int i = 0; i <= level; i++) {
-					if (direction == Constants.DIRECTION.OUT)
-						linkNthLevel.append("(n").append(i).append(":").append(label).append(")").append("-[r")
-								.append(i).append(":connect]->");
-					if (direction == Constants.DIRECTION.IN)
-						linkNthLevel.append("(n").append(i).append(":").append(label).append(")").append("<-[r")
-								.append(i).append(":connect]-");
-					if (direction == null)
-						linkNthLevel.append("(n").append(i).append(":").append(label).append(")").append("-[r")
-								.append(i).append(":connect]-");
-				}
-				String s = (direction == Constants.DIRECTION.IN)
-						? linkNthLevel.substring(0, linkNthLevel.lastIndexOf("[") - 2)
-						: linkNthLevel.substring(0, linkNthLevel.lastIndexOf("[") - 1);
-
-				StringBuilder query = new StringBuilder();
-				query.append("MATCH ").append(s).append(" WHERE n0.id = $UUID ");
-
-				relationProperties.entrySet().forEach(r -> query.append(" AND r").append(level - 1).append(".")
-						.append(r.getKey()).append(" = ").append("'" + r.getValue() + "'"));
-				query.append(" RETURN ");
-				StringBuilder sb = new StringBuilder();
-				if (!CollectionUtils.isEmpty(attributes)) {
-					attributes.stream().forEach(
-							attribute -> sb.append("n").append(level).append(".").append(attribute).append(","));
-					sb.deleteCharAt(sb.length() - 1);
-				} else {
-					sb.append("n").append(level);
-				}
-				query.append(sb).append(" Skip ").append(offset).append(" limit ").append(limit);
-
-				System.out.println("query-> " + query);
-				Statement statement = new Statement(query.toString(), parameters);
-
-				StatementResult result = transaction.run(statement);
-				List<Record> records = result.list();
-
-				transaction.commitAsync().toCompletableFuture();
-				logger.info("Neighbour users for UUID {} found successfully ", UUID);
-				return getNodes(records);
-
-			} catch (ClientException e) {
-				transaction.rollbackAsync().toCompletableFuture();
-				throw new GraphException(ErrorCode.GRAPH_TRANSACTIONAL_ERROR.name(), e.getMessage());
-
-			} finally {
-				transaction.close();
-			}
-		} catch (SessionExpiredException se) {
-			throw new GraphException(ErrorCode.GRAPH_SESSION_EXPIRED_ERROR.name(), se.getMessage());
-		} finally {
-			session.close();
-		}
-
-	}
+    private final Logger logger = LoggerFactory.getLogger(GraphDao.class);
+
+    @Autowired
+    private Driver neo4jDriver;
+
+    private final String label;
+
+    @Autowired
+    public GraphDao(String label) {
+        this.label = label;
+    }
+
+    @Override
+    public Boolean upsertNode(Node node) throws Exception {
+        Session session = neo4jDriver.session();
+        Transaction transaction = session.beginTransaction();
+        try {
+            Statement statement = new Statement("MATCH (n:" + label + ") WHERE n.id=$fromUUID " + "RETURN n", parameters("fromUUID", node.getId()));
```

Comment:
Previous code was using StringBuilder -- which is easy to understand. Do use the same instead of string concat.